static/panes/compiler.ts

```diff
@@ -0,0 +1,3561 @@
+    // we manually set any missing filters to 'false' to recover the old behaviour of
+    // "if it's not here, it's off".
+    _.each(['binary', 'labels', 'directives', 'commentOnly', 'trim', 'intel'], function (oldFilter) {
+        if (filters[oldFilter] === undefined) filters[oldFilter] = false;
+    });
+    return filters;
+}
+
+const languages = options.languages;
+
+type LastResultType = {
+    ppOutput: unknown;
+    astOutput: unknown;
+    irOutput: unknown;
+    devices: unknown;
+    rustMirOutput: unknown;
+    rustMacroExpOutput: unknown;
+    rustHirOutput: unknown;
+    haskellCoreOutput: unknown;
+    haskellStgOutput: unknown;
+    haskellCmmOutput: unknown;
+    gccDumpOutput?: Record<GccDumpOptions, unknown>;
+    gnatDebugTreeOutput: unknown;
+    gnatDebugOutput: unknown;
+    optOutput: unknown;
+};
+
+type CurrentStateType = {
+    id?: number;
+    compiler: string;
+    source: number | null;
+    tree: any;
+    options: unknown;
+    filters: Record<string, unknown>;
+    wantOptInfo: any;
+    libs?: {name: string; ver: string}[];
+    lang: any;
+    selection: monaco.Selection | undefined;
+    flagsViewOpen: any;
+};
+
+type ActiveTools = {
+    id: number;
+    args: any;
+    stdin: string;
+};
+
+type NewToolSettings = {
+    toolId: number;
+    args: any;
+    stdin: string;
+};
+
+type LinkedCodeType = {
+    range: monaco.Range;
+    options: {
+        isWholeLine: boolean;
+        linesDecorationsClassName?: string;
+        className?: string;
+        inlineClassName?: string;
+    };
+};
+
+type Decorations = Record<string, monaco.editor.IModelDeltaDecoration[] | monaco.editor.IModelDeltaDecoration>;
+
+type CompileRequestOptions = {
+    userArguments: string;
+    compilerOptions: {
+        producePp: any;
+        produceAst: boolean;
+        produceGccDump: {
+            opened: boolean;
+            pass?: GccSelectedPass;
+            treeDump?: boolean;
+            rtlDump?: boolean;
+            ipaDump?: boolean;
+            dumpFlags: any;
+        };
+        produceOptInfo: boolean;
+        produceCfg: boolean;
+        produceGnatDebugTree: boolean;
+        produceGnatDebug: boolean;
+        produceIr: boolean;
+        produceLLVMOptPipeline: boolean;
+        produceDevice: boolean;
+        produceRustMir: boolean;
+        produceRustMacroExp: boolean;
+        produceRustHir: boolean;
+        produceHaskellCore: boolean;
+        produceHaskellStg: boolean;
+        produceHaskellCmm: boolean;
+        cmakeArgs?: string;
+        customOutputFilename?: string;
+    };
+    filters: Record<string, boolean>;
+    tools: ActiveTools[];
+    libraries: {
+        id: string;
+        version: string;
+    }[];
+};
+
+type CompileRequest = {
+    source: string;
+    compiler: string;
+    options: CompileRequestOptions;
+    lang: any;
+    files: FiledataPair[];
+    bypassCache: boolean;
+};
+
+type Assembly = {
+    labels?: any[];
```

Comment:
I think we might have a proper type for this in the backend, I'd actually be surprised if we don't have the whole `Assembly` type already

Reply:
I didn't find a type called `Assembly` (or similar), maybe it's called completely different?